docusaurus/docs/iOS/guides/working-with-messagelist-layout.md

```diff
@@ -0,0 +1,128 @@
+         // If you want, you can include the author name for the message as well.
+         options.insert(.authorName)
+         return options
+     }
+ }
+ ```
+ 
+ When your custom `ChatMessageLayoutOptionsResolver` is created, all you need to do is to set this class to `Components` before displaying the `MessageList`:
+ 
+ ```swift
+ Components.default.messageLayoutOptionsResolver = LeftAlignedMessageLayoutOptionsResolver()
+ ```
+ Here is the result: 
+ 
+ ![](../assets/messagelist-layout-left-alignment.png) 
+ 
+## Hiding bubbles
+
+If you need to hide only the bubbles, we suggest you to implement custom subclass of  `MessageLayoutOptionsResolver`  as well. 
+After that all you need to do is to remove the `bubble` option.
+
+```swift
+
+class NoBubblesMessageLayoutOptionsResolver: ChatMessageLayoutOptionsResolver {
+    override func optionsForMessage(at indexPath: IndexPath, in channel: _ChatChannel<NoExtraData>, with messages: AnyRandomAccessCollection<_ChatMessage<NoExtraData>>) -> ChatMessageLayoutOptions {
+        // Get options for the message at given indexPath to change it.
+        var options = super.optionsForMessage(at: indexPath, in: channel, with: messages)
+        options.remove(.bubble)
+        return options
+    }
+}
+``` 
+
+When your custom `ChatMessageLayoutOptionsResolver` is created, all you need to do is to set this class to `Components` before displaying the `MessageList`:
+
+```swift
+Components.default.avatarView = NoBubblesMessageLayoutOptionsResolver()
+```
+
+![](../assets/messagelist-layout-nobubbles.png)
+
+## Disabling grouping of messages
+The default behaviour of `ChatMessageLayoutOptionsResolver` is that it checks the messages if they are grouped or not. 
+For this being possible we have created `isLastInSequence` property which is used for grouping the messages. 
+
+```swift
+...
+// Check if the messages is sent last to create continuous bubble effect.
+if !isLastInSequence {
+    options.insert(.continuousBubble)
+}
+// For current user, add Padding to avatar
+if !isLastInSequence && !message.isSentByCurrentUser {
+    options.insert(.avatarSizePadding)
+}
+// And to make the group effect, let's add timestamp to the bottom.
+if isLastInSequence {
+    options.insert(.timestamp)
+}
+...
+```
+
+So if you want to disable grouping of messages, just subclass `ChatMessageLayoutOptionsResolver` and do not use `isLastInSequence`.  
+
+## Square avatars
+
+First subclass `ChatAvatarView` and set it according to your needs 
+
+```swift
+final class SquareAvatarView: ChatAvatarView {
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        imageView.layer.cornerRadius = 3
+    }
+}
+``` 
+
+After doing that, you need to set this custom view to `Components` somewhere where your customisation takes place.. 
+
+```swift
+Components.default.avatarView = SquareAvatarView.self
+```
+![](../assets/messagelist-layout-square-avatars.png)
+
+## Moving components of Messages in the layout
```

Comment:
Need more examples in this section.
- How to achieve this layout:
```
Avatar - Name - Date
Message
```
- How to move certain elements in the layout? The guide should give an overview about which elements belong to which container. A user trying to customize the layout doesn't know that name field is named `authorNameLabel` and is contained inside `metadataContainer`, which is contained inside `bubbleThreadMetaContainer`. This should be like reference docs, but something like an overview image, so when a user is trying to change position of X element, they know which containers to touch.

Reply:
I generated picture with some simple description as well as overview image, you can try to run the documentation.